aten/src/ATen/native/Convolution.cpp

```diff
@@ -696,7 +705,15 @@ at::Tensor _convolution(
     }
 #endif
   } else if (input.device().type() == c10::DeviceType::CPU || input.device().type() == c10::DeviceType::CUDA) {
```

Comment:
I'm confused.  We're doing a "use_xnnpack" check when the device type is CUDA?

Reply:
The check is if device type is CPU or CUDA, right?  use_cpu_depthwise3x3_winograd in this block is also CPU only.  I can break the check into two sections if you prefer.

Reply:
Yeah.  This code makes it look like use_xnnpack or use_cpu_depthwise3x3_winograd might return true on a CUDA input.  I think a clearer option would be
```
} else if (is_cpu && use_xnn) {
  ...
} else if (is_cpu && use_winograd) {
  ...
} else if (is_cpu || is_cuda) {
  ...
}
```

Or even drop the `DeviceType::CPU` checks and move them into `use_xnnpack` and `use_winograd`

Reply:
Sounds good, will change.